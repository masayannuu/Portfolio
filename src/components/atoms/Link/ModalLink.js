import React from 'react'
import styled from 'styled-components'
import { Icon, Field } from 'reactbulma'

const StyledField = styled(Field)`
  margin-right: 0.8rem;
  justify-content: flex-end !important;
  font-size: 1rem;
  position: relative;
  &:hover {
    cursor: pointer;
    color: #a2c4c7;
  }
`

const ModalLink = ({ onClick }) => (
  <StyledField grouped onClick={onClick}>
    <Icon>
      <i className="fa fa-angle-right" />
    </Icon>
    Click & More
  </StyledField>
)

export default ModalLink
