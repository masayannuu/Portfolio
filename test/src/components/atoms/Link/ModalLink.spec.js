import React from 'react'
import Enzyme from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import ModalLink from '../../../../../src/components/atoms/Link/ModalLink'

describe('Link ModalLink Atoms', () => {
  it('should render ModalLink', () => {
    const wrapper = Enzyme.shallow(<ModalLink onClick={() => console.log('test')} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
