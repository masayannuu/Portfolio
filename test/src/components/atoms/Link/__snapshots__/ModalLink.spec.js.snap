// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link ModalLink Atoms should render ModalLink 1`] = `
<ModalLink__StyledField
  grouped={true}
  onClick={[Function]}
>
  - Click & More
</ModalLink__StyledField>
`;
